services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -1122,8 +1124,9 @@ async def dispatch(self, request, call_next):  # pylint: disable=too-many-branch
 
         response = await call_next(request)
 
-        # Don't forward responses that fail
-        if response.status_code != 200:
+        # Don't forward responses that fail.
+        # NOTE: the 30x (redirect responses) are use by the oauth2 authentication.
+        if response.status_code not in (HTTP_200_OK, HTTP_302_FOUND, HTTP_307_TEMPORARY_REDIRECT):
             if response is None:
                 return None
```

Comment:
Non related to this PR, should dispatch return other types than JSONResponse / HTTPException?

Reply:
I don't know, but in my case I'm returning a RedirectResponse and it's working well